lib/ui/home/tabs/hadeth_tab/hadeth_tab.dart

```diff
@@ -1,12 +1,73 @@
 import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:islami_c11_thursday/ui/home/tabs/hadeth_tab/hadeth_title_widget.dart';
+import 'package:islami_c11_thursday/utils/image_utils.dart';
 
-class HadethTab extends StatelessWidget {
-  const HadethTab({super.key});
+class HadethTab extends StatefulWidget {
+  @override
+  State<HadethTab> createState() => _HadethTabState();
+}
+
+class _HadethTabState extends State<HadethTab> {
+  List<HadithItem> allHadithList = [];
 
   @override
   Widget build(BuildContext context) {
-    return Container(
-      color: Colors.greenAccent,
-    );
+    if (allHadithList.isEmpty) loadHadithFile();
+    return allHadithList.isEmpty
+        ? Center(
+            child: CircularProgressIndicator(),
+          )
+        : Container(
+            child: Column(
+              children: [
+                Expanded(
+                    flex: 1,
+                    child: Image.asset(
+                        getImagePathByName('hadeth_header_image.png'))),
+                Expanded(
+                  child: Expanded(
+                    flex: 2,
+                    child: ListView.separated(
+                        itemBuilder: (context, index) => HadithTitleWidget(
+                            hadithTitle: allHadithList[index].title),
+                        separatorBuilder: (context, index) => Container(
+                              margin: EdgeInsets.symmetric(horizontal: 26),
+                              color: Color(0xFFB7935F),
+                              width: double.infinity,
+                              height: 2,
+                            ),
+                        itemCount: allHadithList.length),
+                  ),
+                )
+              ],
+            ),
+          );
   }
+
+  void loadHadithFile() async {
```

Comment:
refactor this function to improve code redability
changet name to apply naming convention